runtime/oti/j9nonbuilder.h

```diff
@@ -5773,6 +5773,16 @@ typedef struct J9JavaVM {
 #else
 	omrthread_monitor_t classUnloadMutex;
 #endif
+	/* When GC runs as part of class redefinition, it could unload classes and
+	 * therefore try to acquire the class unload mutex, which is already held
+	 * to prevent JIT compilation from running concurrently with redefinition.
+	 * Such an acquisition attempt could deadlock if it occurs on a different
+	 * thread, even though the acquisition would be morally recursive.
```

Comment:
Perhaps "morally" should be "normally" here?

Reply:
An acquisition that occurs on a different thread is technically never a recursive acquisition. However, in this case it's tantamount to one, and the fact that it's occurring on another thread is only a technicality. So I meant *morally* recursive as opposed to *technically* recursive